pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -147,67 +147,47 @@ func watchChaosResources(clientSet client.Client, c controller.Controller) error
-	// If thats the case, then udpate the engineStatus to completed
-	if isCompleted && engine.Instance.Status.EngineStatus != litmuschaosv1alpha1.EngineStatusCompleted {
-		if err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusCompleted); err != nil {
-			r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to update chaosengine")
-			return reconcile.Result{}, err
-		}
-		r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "ChaosEngineCompleted", "Chaos Engine completed, will delete or retain the resources according to jobCleanUpPolicy")
+	// Start the reconcile by setting default values into Chaos Engine
+	if err := r.initEngine(engine); err != nil {
+		return reconcile.Result{}, err
 	}
 
-	// Verify that the engineStatus is set to completed,
-	// if thats the case, then return reconcileForComplete
-	if checkEngineStatusForComplete(engine) {
-		return r.reconcileForComplete(request)
+	// Handling of normal execution of Chaos Engine
+	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateActive && engine.Instance.Status.EngineStatus == litmuschaosv1alpha1.EngineStatusInitialized {
+		return r.reconcileForCreationAndRunning(engine, reqLogger)
 	}
 
-	// Verify that the engineState is set to stop
-	// if true, set the engineStatus as stopped
-	if checkEngineStateForStop(engine) {
-		if err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusStopped); err != nil {
-			r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to update chaosengine")
-			return reconcile.Result{}, err
-		}
+	// Handling Graceful completion of Chaos Engine
+	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateStop && engine.Instance.Status.EngineStatus == litmuschaosv1alpha1.EngineStatusCompleted {
+		return r.reconcileForComplete(request)
 	}
 
-	// Verify that the engineStatus is set to stopped,
-	// then return reconcileForDelete
-	if checkEngineStatusForStopped(engine) {
+	// Handling forceful Abort of Chaos Engine
+	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateStop && engine.Instance.Status.EngineStatus != litmuschaosv1alpha1.EngineStatusCompleted {
 		return r.reconcileForDelete(request)
 	}
 
-	// Verify that the engineState, and engineStatus to initalized chaos engine resources
-	if checkEngineForCreation(engine) {
-		if err = r.checkRunnerPodForCompletion(engine, reqLogger); err != nil {
-			r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to update chaosengine")
-			return reconcile.Result{}, err
-		}
-		return r.reconcileForCreationAndRunning(engine, request)
+	// Handling restarting of Chaos Engine
+	if engine.Instance.Spec.EngineState == litmuschaosv1alpha1.EngineStateActive && engine.Instance.Status.EngineStatus != litmuschaosv1alpha1.EngineStatusInitialized {
+		return r.reconcileForRestart(request)
```

Comment:
Like discussed, can we add EngineStatusCompleted for more clarity over != EngineStatusInitialized ?

Reply:
Yes, it the latter case, but still will add them both separately.